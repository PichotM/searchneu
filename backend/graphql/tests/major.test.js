import { createTestClient } from 'apollo-server-testing';
import { gql } from 'apollo-server';
import server from '../index';
import db from '../../../models/index';

const { query } = createTestClient(server);
const MajorData = db.MajorData;


beforeAll(async () => {
  await MajorData.create({
    majorId: 'computer-information-science/computer-science/bscs',
    catalogYear: 2018,
    name: 'Computer Science',
    requirements: { name: 'Computer Science, BSCS', yearVersion: 2018 },
    plansOfStudy: [ { years: [ 1000 ] , id: '0' } ],
  });

  await MajorData.create({
    majorId: 'computer-information-science/computer-science/bscs',
    catalogYear: 2017,
    name: 'Computer Science',
    requirements: { name: 'Computer Science, BSCS', yearVersion: 2017 },
    plansOfStudy: [ { years: [ 1000 ], id: '0' } ],
  });

  await MajorData.create({
    majorId: 'science/biochemistry/biochemistry-bs',
    catalogYear: 2018,
    name: 'Biochemistry',
    requirements: { name: 'Biochemistry, BS', yearVersion: 2018 },
    plansOfStudy: [ { years: [ 1000 ], id: '0' } ],
  });
});

it('gets major from majorId', async () => {
  const res = await query({
    query: gql`
      query major {
        major(majorId: "computer-information-science/computer-science/bscs") {
          name
        }
      }
    `;
  });
  expect(res).toMatchSnapshot();
});

it('gets specific occurrence', async () => {
  const res = await query({
    query: gql`
      query major {
        major(majorId: "computer-information-science/computer-science/bscs") {
          name
          occurrence(year: 2017) {
            catalogYear
            requirements
            plansOfStudy
          }
        }
      }
    `;
  });
});

it('gets latest occurrence', async () => {
  const res = await query({
    query: gql`
      query major {
        major(majorId: "computer-information-science/computer-science/bscs") {
        name
        latestOccurrence {
          catalogYear
          requirements
        }
        }
      }
    `;
  });
});
