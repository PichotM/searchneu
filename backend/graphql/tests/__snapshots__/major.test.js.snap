// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cannot find major from non-present majorId 1`] = `
Object {
  "data": Object {
    "major": null,
  },
  "errors": Array [
    [GraphQLError: connect ECONNREFUSED 127.0.0.1:5432],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`cannot find majorOccurrence from non-present year 1`] = `
Object {
  "data": Object {
    "major": null,
  },
  "errors": Array [
    [GraphQLError: connect ECONNREFUSED 127.0.0.1:5432],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`gets latest occurrence 1`] = `
Object {
  "data": Object {
    "major": null,
  },
  "errors": Array [
    [GraphQLError: connect ECONNREFUSED 127.0.0.1:5432],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`gets major from majorId 1`] = `
Object {
  "data": Object {
    "major": null,
  },
  "errors": Array [
    [GraphQLError: connect ECONNREFUSED 127.0.0.1:5432],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`gets specific occurrence 1`] = `
Object {
  "data": Object {
    "major": null,
  },
  "errors": Array [
    [GraphQLError: connect ECONNREFUSED 127.0.0.1:5432],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
